pandas/core/indexing.py

```diff
@@ -118,7 +118,7 @@ def __getitem__(self, key):
             key = tuple(com.apply_if_callable(x, self.obj) for x in key)
             try:
                 values = self.obj._get_value(*key)
-            except (KeyError, TypeError):
+            except (KeyError, TypeError, InvalidIndexError):
```

Comment:
can you add a comment about why this is caught here

Reply:
Do we need to test this somehow as well?

Reply:
See https://github.com/pandas-dev/pandas/pull/27259 for a discussion on that. I can add a test with the code snippet I show there, but is a rather dummy example ..

Reply:
I guess this is fine, though why are you not using ``.loc`` and *using* ``__getitem__``?

Reply:
> why are you not using `.loc` and using `__getitem__`? 

This is from a custom Indexer attribute in geopandas, so it is on purpose not using `.loc`. Now, I don't say it is the best implementation by using such pandas internals, opened an issue about that in geopandas https://github.com/geopandas/geopandas/issues/1042